src/irmin-pack/unix/lower.ml

```diff
@@ -207,7 +207,27 @@ module Make_volume (Io : Io.S) (Errs : Io_errors.S with module Io = Io) = struct
         new_control
     in
     let* () = Control.close c in
-    Ok root
+    Ok (identifier t)
+
+  let swap ~generation t =
+    let root = path t in
+    let control_tmp =
+      Irmin_pack.Layout.V5.Volume.control_gc_tmp ~generation ~root
+    in
+    let control = Irmin_pack.Layout.V5.Volume.control ~root in
+    Io.move_file ~src:control_tmp ~dst:control
+
+  let cleanup ~generation t =
+    let clean filename =
+      match Irmin_pack.Layout.Classification.Volume.v filename with
+      | `Control_tmp g when g = generation -> swap ~generation t
+      | `Control_tmp g when g != generation ->
```

Comment:
Shouldn't that be `<>` instead of `!=`? Though in this case it should work similarly.

Reply:
Oh yes -- thanks! Old habits die hard. :sweat_smile: